tools/nl/common_name_entity/main.py

```diff
@@ -0,0 +1,163 @@
+def _get_chunked_names(names):
+  chunked_names = []
+  curr_chunk_names = []
+  curr_char_length = 0
+  for name in names:
+    if len(name) + curr_char_length + 1 > _MAX_SPACY_CHARS:
+      chunked_names.append(curr_chunk_names)
+      curr_char_length = 0
+    curr_char_length += len(name) + 1
+    curr_chunk_names.append(name)
+  chunked_names.append(curr_chunk_names)
+  return chunked_names
+
+
+# Gets the flagged tokens and flagged names for a chunk of names
+# flagged_tokens is a map of token to its part of sentence for tokens that are
+# flagged as possibly being a common word
+# flagged_names is a map of name to its entity type for names where all its
+# tokens are flagged
+def _get_results_for_names(nlp, names, name_to_type):
+  if not names:
+    return {}, {}
+  name_str = ' '.join(names)
+  doc = nlp(name_str)
+  # map of name to its list of tokens that make up the name
+  name_to_tokens = {}
+  # index in names list of the current name we are processing
+  curr_name_idx = 0
+  # the part of the current name that has yet to be processed
+  running_curr_name = names[curr_name_idx]
+  flagged_tokens = {}
+  for token in doc:
+    # If running_curr_name is empty, we've finished getting the tokens for
+    # this name, so move on to the next name
+    if not running_curr_name and curr_name_idx < len(names) - 1:
+      curr_name_idx += 1
+      running_curr_name = names[curr_name_idx]
+    # If running_curr_name starts with the token text, that means this token
+    # applies to this name. Add the token to name_to_tokens and update
+    # running_curr_name. We need a running_curr_name because spacy will tokenize
+    # a single name into multiple tokens and we need to match the name back to
+    # its tokens.
+    # e.g., if we have the names "brilliant blue" and "me" and we pass the nlp
+    # "brilliant blue me", we will get back the results for the tokens
+    # "brilliant", "blue", and "me". The first running_curr_name will be
+    # "brilliant blue" & after processing the first token, running_curr_name
+    # will be "blue", and then it will be "" to signify we've processed all the
+    # tokens for "brilliant blue", and can move on to the next name.
+    if running_curr_name.startswith(token.text):
+      curr_name = names[curr_name_idx]
+      if not curr_name in name_to_tokens:
+        name_to_tokens[curr_name] = []
+      name_to_tokens[curr_name].append(token.text)
+      running_curr_name = running_curr_name[len(token.text):].strip()
+    # Add token to flagged tokens if it is not in _POS_GOOD (_POS_GOOD are part
+    # of sentence tags that are likely to not be a common word) and it is not a
+    # combination of letters and numbers.
+    if token.pos_ not in _POS_GOOD and re.match(_LETTER_NUMBER_REGEX,
+                                                token.text) is None:
+      flagged_tokens[token.text] = token.pos_
+  flagged_names = _generate_flagged_names(flagged_tokens, name_to_tokens,
```

Comment:
nit: some vertical spaces for readability?

Reply:
My silly rule of thumb:  if there's a block of comments, leave a space above it, and after the code block that describes the comments.

Reply:
done